### PR TITLE
Add Deploy tokens API

### PR DESCRIPTION
**Pull request by dylanmtaylor**
Reference: https://docs.gitlab.com/ee/api/deploy_tokens.html

### CODE REVIEW

**Comment by dylanmtaylor** [comment]
I'm trying to implement the deploy tokens API. Could someone review this and provide feedback? I think this is the right way to go about implementing it.

**Comment by jetersen** [comment]
@dylanmtaylor look like a failed commit here: https://github.com/jdalrymple/gitbeaker/pull/762/commits/09fb95068c9ccdd84ce78005a27b63c2f3b7e76e

**Comment by jdalrymple** [comment]
Just rebase on the latest master, and it should be good to go

**Comment by dylanmtaylor** [comment]
I think I fixed it - I don't see unrelated files changed under https://github.com/jdalrymple/gitbeaker/pull/762/files

File: packages/gitbeaker-core/src/templates/ResourceDeployTokens.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  all({ resourceId, ...options }: { resourceId?: string | number } & PaginatedRequestOptions) {
+
+    if (resourceId) {
+      url = `${encodeURIComponent(resourceId)}/deploy_tokens`;
+    } else {
+      // https://docs.gitlab.com/ee/api/deploy_tokens.html#list-all-deploy-tokens (FIXME?) This needs to be at the root of the webserver, not after the resource type
+      url = 'deploy_tokens';
+    }
+
```
Comment (jdalrymple) [inline packages/gitbeaker-core/src/templates/ResourceDeployTokens.ts:44]: This is consistent with other API endpoints. There is no need to do the join below i don't think, and you probably should define this variable

**Comment by jdalrymple** [comment]
LGTM, @jetersen you see anything else?

**Comment by dylanmtaylor** [comment]
If you think it's all good, feel free to merge it in :)

**Comment by jetersen** [comment]
Perhaps update the PR title?

**Comment by jdalrymple** [comment]
:rocket: PR was released in [`19.4.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/19.4.0) :rocket:

**Pull request merged**